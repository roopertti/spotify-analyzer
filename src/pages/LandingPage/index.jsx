import React from 'react';
import styled from 'styled-components';

import Layout from '../../components/common/Layout';
import PrimaryButton from '../../components/common/PrimaryButton';

/* eslint-disable no-console */
function initAuth() {
  console.log('Test');
}

const Section = styled.section`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  padding: 5rem 1rem;

  & > h1 {
    font-size: 1.5rem;
  }

  & > h5 {
    font-size: 1rem;
    color: #343434;
    margin-bottom: 3rem;
  }

  @media (min-width: 960px) {
    & > h1 {
      font-size: 2.5rem;
    }

    & > h5 {
      color: #343434;
      font-size: 1.5rem;
      margin-bottom: 5rem;
    }
  }

  @media (min-width: 600px) {
    & > h1 {
      font-size: 2rem;
    }

    & > h5 {
      color: #343434;
      font-size: 1.1rem;
      margin-bottom: 4rem;
    }
  }
`;

export default function LangingPage() {
  return (
    <Layout>
      <Section>
        <h1>Authorize with your Spotify account to analyze tracks.</h1>
        <h5>Authorization is required to access your playlists.</h5>
        <PrimaryButton onClick={initAuth}>Authorize</PrimaryButton>
      </Section>
    </Layout>
  );
}
