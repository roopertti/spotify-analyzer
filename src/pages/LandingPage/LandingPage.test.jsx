import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import LandingPage from './index';

describe('LandingPage tests', () => {
  it('Render layout', () => {
    const tree = renderer.create(<LandingPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
