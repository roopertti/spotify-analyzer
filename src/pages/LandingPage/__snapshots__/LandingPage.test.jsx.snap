// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage tests Render layout 1`] = `
Array [
  <header
    className="sc-gsTCUz exgbLI"
  >
    <h1
      className="sc-bdfBwQ fgPLm"
    >
      Spotify analyzer
    </h1>
    <button
      className="sc-dlfnbm fNZAuq"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-caret-down fa-w-10 "
        data-icon="caret-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </header>,
  <div
    className="sc-hKgILt gfXTHO"
  >
    <section
      className="sc-pFZIQ iJjGaL"
    >
      <h1>
        Authorize with your Spotify account to analyze tracks.
      </h1>
      <h5>
        Authorization is required to access your playlists.
      </h5>
      <button
        className="sc-fubCfw iEiSWV"
        onClick={[Function]}
      >
        Authorize
      </button>
    </section>
  </div>,
  <footer
    className="sc-gKsewC kxsEIp"
  >
    <p
      className="sc-iBPRYJ hSzIJK"
    >
      Created by Robert Kuhlmann
    </p>
    <a
      className="sc-eCssSg jkrLHO"
      href="#"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-github fa-w-16 sc-jSgupP hdqZzO"
        data-icon="github"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Source code
      </span>
    </a>
  </footer>,
]
`;
